legal-api/src/legal_api/models/business.py

```diff
@@ -563,6 +564,21 @@ def get_filing_by_id(cls, business_identifier: int, filing_id: str):
             one_or_none()
         return None if not filing else filing[1]
 
+    @classmethod
+    def is_pending_amalgamating_business(cls, business_identifier: int):
+        """Check if a business has a pending amalgamation with the provided business identifier. """
+        # Construct a JSON containment check clause for the SQL query
+        where_clause = text(
+            "filing_json->'filing'->'amalgamationApplication'->'amalgamatingBusinesses'" +
+            f' @>\'[{{"identifier": "{business_identifier}"}}]\'')
+        # pylint: disable=protected-access
+        filing = db.session.query(Filing). \
+        filter(Filing._status == Filing.Status.PAID.value,
+               Filing._filing_type == 'amalgamationApplication',  # Check for the filing type
+               where_clause  # Apply the JSON containment check
+               ).one_or_none()
```

Comment:
As mentioned in our conversation, see if this more pythonic way of querying works:

```python
# Construct a JSON containment check clause for the SQL query
    where_clause = { "identifier": business.identifier }

    # Query the database to find amalgamation filings
    filing = db.session.query(Filing). \
        filter(and_(Filing._status == Filing.Status.PAID.value,
                    Filing._filing_type == AMALGAMATION_APPLICATION,  # Check for the filing type
                    Filing.filing_json['filing']['amalgamationApplication']['amalgamatingBusinesses'].contains(where_clause)  # Apply the JSON containment check
                    )).one_or_none()
```